hirundo/_http.py

```diff
@@ -7,7 +7,10 @@
 
 def raise_for_status_with_reason(response: Response):
     try:
-        response.reason = response.json().get("reason", None)
+        if response.status_code >= 400:
```

Comment:
If the status is smaller than 400, then there is no error, so there's no point in parsing the body for an error which won't ever be thrown